src/main/java/dev/tycho/stonks/scheduledtasks/SubscriptionCheckTask.java

```diff
@@ -0,0 +1,112 @@
+import org.bukkit.entity.Player;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public class SubscriptionCheckTask implements Runnable {
+
+  public SubscriptionCheckTask() {
+    System.out.println("New subscription payment task scheduled");
+  }
+
+  @Override
+  public void run() {
+    System.out.println("Automatically renewing subscriptions");
+    //We want to go through all subscriptions
+    List<Subscription> subscriptions;
+    try {
+      subscriptions = DatabaseHelper.getInstance().getDatabaseManager().getSubscriptionDao().queryForAll();
+    } catch (SQLException e) {
+      e.printStackTrace();
+      Bukkit.broadcastMessage("Error updating subscriptions");
+      return;
+    }
+
+    for (Subscription subscription : subscriptions) {
+      renewSubscription(subscription);
+      cancelSubscriptionIfOverdue(subscription);
+    }
+  }
+
+  private void renewSubscription(Subscription subscription) {
+    Service service = subscription.getService();
+    OfflinePlayer player = Bukkit.getOfflinePlayer(subscription.getPlayerId());
+    Player onlinePlayer = null;
+    if (player.isOnline()) onlinePlayer = player.getPlayer();
+
+    if (subscription.isOverdue() && subscription.isAutoPay()) {
+      System.out.println("Auto billing " + player.getName() + " $" + service.getCost() + " for service id " + service.getId());
+      //Subscription is overdue and we must renew it
+      if (onlinePlayer != null)
+        onlinePlayer.sendMessage(ChatColor.YELLOW + "Your subscription for " + service.getName() + " has ended." + ChatColor.GREEN + " Automatically renewing it now.....");
+
+      //Try and charge them the amount due
+      if (!Stonks.economy.withdrawPlayer(player, service.getCost()).transactionSuccess()) {
+        //Player does not have enough money
+        System.out.println("(failed - they didn't have enough money)");
+        if (onlinePlayer != null)
+          onlinePlayer.sendMessage(ChatColor.RED + "Failed to renew. You don't have enough money, please get more then manually resubscribe");
+      } else {
+        //Payment success
+        //Update that the subscription is paid
+        subscription.registerPaid();
+        try {
+          //Update the subscription in the database
+          DatabaseHelper.getInstance().getDatabaseManager().getSubscriptionDao().update(subscription);
+
+          //Pay and update the account
+          service.getAccount().getAccount().addBalance(service.getCost());
+
+          //Update the account in the database and add a log
+          DatabaseHelper.getInstance().getDatabaseManager().updateAccount(service.getAccount().getAccount());
+          DatabaseHelper.getInstance().getDatabaseManager().logTransaction(new Transaction(service.getAccount(),
+              player.getUniqueId(), "Subscription payment for " + service.getName(), service.getCost()));
+
+          if (onlinePlayer != null) {
+            onlinePlayer.sendMessage(ChatColor.GREEN +
```

Comment:
You should copy the `sendMessage(Player, String)` method to here and use it for this

Reply:
https://github.com/schlatt-co/stonks/blob/a34c87a8a16fdce21ba0462044aedb07a35af01b/src/main/java/dev/tycho/stonks/scheduledtasks/SubscriptionCheckTask.java#L113